proposed/dotnet-bootstrapper/dotnet-bootstrapper-path.md

```diff
@@ -0,0 +1,130 @@
+# Integrating a 'dotnet' Bootstrapper with Visual Studio (PATH Issues)
+
+## Introduction
+
+How can we enable an `nvm` or `rustUp` like experience for `dotnet`? In this document, the command to do so will be aliased as `bootstrapper`, and we will only discuss PATH concerns. This may be `dotnet` going forward, or something else.
+
+One of the core differences in our ecosystem compared to other toolchains is the inclusion of .NET in VS. Visual Studio and the MSI or 'global' .NET Installer set the System `PATH` on Windows and not the user level `PATH`. The system `PATH` always takes precedence over the user `PATH`, which is how other toolchains manage user local install folders. Below are proposals to remedy this situation.
+
+Please see the `Technical Details` section for more context.
+
+## Proposals
+
+### Update the Installers / Visual Studio
+
+:star: A framing document and prototype made by the .NET SDK team relies on and suggests this proposal. We believe this to be the best proposal.
+
+One option is to set the machine into a 'local' SDK mode that users could opt into when users engage with `bootstrapper` to do their first local install. This 'mode' would tell the system-wide installers (both `pkg` and `msi`) to not set the system `PATH`.
+
+In this scenario:
+
+1. When the user interfaces with the `bootstrapper`, to install an SDK, they are prompted about the change to `local` installs. They are also informed about the impact this has on VS and can pick an update cadence.
+
+2. On Windows, a registry key is set in `HKLM` (the registry hive for local user contexts). `bootstrapper` must elevate to enable this registry edit when prompting the user.
+
+> 📓 Technical Details:
+> A registry key is preferred over an environment variable because it is an admin-protected context. We don't want user-local controllable behavior to influence/break the behavior of an administrative installer, as that is a security concern. A policy key may also be used, but a policy key is essentially equivalent to a registry key in this context.
+>
+> On Unix, we could set this via a protected sentinel file.
+
+3. Under step 2, the `PATH` is updated to a dotnet application host (dotnet.exe) that contains the local installs managed by `bootstrapper`. Visual Studio and the other installers will no longer set the System `PATH` until the registry key or sentinel change is reverted. The option to undo this would be available, by removing the key, perhaps via an `uninstall all` or `reset` CLI feature from `bootstrapper`. The exact semantics of such are not pertinent here.
+
+4. The existing `global.json` `paths` feature would still enable other scenarios like for Visual Studio to specify any SDK. In this sense, it is already possible to enable local .NET SDKs to work with VS.
```

Comment:
This seems a bit unclear to me.  The `paths` feature would still work, but it doesn't let Visual Studio specify an SDK.

Reply:
Ah, I think I had meant to enable VS to use an 'admin' SDK rather than a local one.

Reply:
Which doesn't work very well due to the lack of env variables explained below